### PR TITLE
feat: implement screwdriver helm chart

### PR DESCRIPTION
**Pull request by minzcmu**
- to allow user to set up screwdriver in k8s easily
- only secrets file need to be applied manually

### CODE REVIEW

File: templates/api.yaml [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+  ports:
+  - port: 80
+    targetPort: 80
+  selector:
+    app: api
+    env: {{ .Values.env }}
+    tier: app
+---
+# requires service account to be created
+apiVersion: extensions/v1beta1
```
Comment (tk3fftk) [inline templates/api.yaml:20]: Should we use this old api version for Deployment?

Comment (minz1027) [reply templates/api.yaml]: looks like for versions before k8s 1.9, we cannot use `app/v1` :(  Not sure if we should make any requirements or assumptions on what k8s version they need to be on.
https://kubernetes.io/docs/concepts/workloads/controllers/deployment/

Comment (minzcmu) [reply templates/api.yaml:20]: Thanks for reviewing this huge PR! lol 

Looks like for versions before k8s 1.9, we cannot use `app/v1` :(  Not sure if we should make any requirements or assumptions on what k8s version they need to be on.
https://kubernetes.io/docs/concepts/workloads/controllers/deployment/

File: generate_secrets.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+cp screwdriver-api-secrets.tmpl screwdriver-api-secrets.yaml
+
+for name in '{{cookiepassword}}' '{{encryptionpassword}}' '{{hashingpassword}}'
+do
+    value=`cat /dev/urandom | env LC_CTYPE=C tr -dc a-zA-Z0-9 | head -c 32 | base64`
```
Comment (pranavrc) [inline generate_secrets.sh:12]: Does this need to be inside the loop?

Comment (minzcmu) [reply generate_secrets.sh:12]: this is generating 3 random values. put it in the loop to avoid doing that 3 times

Comment (minzcmu) [reply generate_secrets.sh:12]: this is generating 3 random values. put it in the loop to avoid doing that 3 times

**Review by jithine — APPROVED** [review]

**Review by pranavrc — APPROVED** [review]
image

**Pull request merged**